src/binary_heap.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals, division
+
+
+class BinaryHeap(object):
+    """Binary Heap data structure class object"""
+    def __init__(self, iterable=None):
+        """Init an instance with the option to pass in an iterable"""
+        self._list = []
+        try:
+            for element in iterable:
+                self._list.append(element)
+            self._list.sort()
+        except TypeError:
+            if iterable is not None:
+                self._list.append(iterable)
+
+    def _find_parent_index(self, child_index):
+        """Returns parent's index of a value given its index"""
+        return (child_index - 1) // 2 
+
+    def _find_smaller_child_index(self, parent_index):
+        """Returns a value's smaller child's index given its index,
+        returns left child's index if right child does not exist"""
+        left_child_index = parent_index * 2 + 1
+        left_child_value = self._list[left_child_index]
+        right_child_index = parent_index * 2 + 2
+        try:
+            right_child_value = self._list[right_child_index]
+        except IndexError:
+            return left_child_index
+        return left_child_index if left_child_value <=\
+            right_child_value else right_child_index
+
+    def _swap_indexes(self, index1, index2):
+        """Helper function that swap 2 indexes of _list"""
+        self._list[index1], self._list[index2] =\
```

Comment:
Suggestions on how you can fit this into one line and avoid the ugly backslash line continuation char:

reduce the length of the `index1` and `index2` names
set `self._list` to a shorter name, then use that e.g.
```
heap = self._list
heap[ix1], heap[ix2] ...
```